assets.src/src/redmine-tracky/controllers/timer-controller.ts

```diff
@@ -56,18 +55,26 @@ export default class extends Controller {
         }
     }
 
-    private timeDifference(): string {
-        const startDateTime = this.convertToDateTime((this.startTarget as HTMLInputElement).value);
-        const endDateTime = this.convertToDateTime((this.endTarget as HTMLInputElement).value);
+    private timeDiffToString(timeDiff: TimeDiff) {
+        const sign = (timeDiff.minutes < 0 || timeDiff.seconds < 0) ? '-' : '';
+        const [hours, mins, secs] = [timeDiff.hours, timeDiff.minutes, Math.round(timeDiff.seconds)]
+                                    .map((value) => value.toString().replace('-', '').padStart(2, '0'));
+        const hoursFormat = Number(hours) > 0 ? `${hours}:` : '';
+    
+        return `${sign}${hoursFormat}${mins}:${secs}`;
+    }
```

Comment:
let's go fancy and chose a functional approach: 

```suggestion
    private timeDiffToString(duration: Duration): string {
        const negativeDuration = duration.as('milliseconds') < 0;
        const sign = negativeDuration ? '-' : '';
        return sign.concat(Object.values(duration.toObject())
          .filter((x, index) => !(index == 0 && !x)) // Remove leading units
          .map(value => Math.round(value))
          .map(value => Math.abs(value)) // Remove negative sign
          .map(value => value.toString().padStart(2, '0'))
          .join(':'));
    }
```

Reply:
please test your code before suggesting it. you are mapping the `sign` too.

Reply:
why did you change the `timeDiff` param to `duration`? then you'd have to change the code elsewhere